### ISSUE TITLE
[Feat]: Graph Panel + support for non time series data 

### ISSUE THREAD

**Issue by torkelo** [grafana/grafana#7158]
Graph panel should support json doc and table data formats.

**Comment by towolf** [comment]
But this kinda works already when switching X-axis from Mode = Time to Mode = Series?

My only problem with that is, that this is kind of bugged with ES and removing the 'Date Histogram' Group-by.  

I want the total percentiles in the set time range, not percentiles of date histogram bins.

Going from this

![image](https://cloud.githubusercontent.com/assets/345210/21956939/b25f38c6-da8b-11e6-86af-66535d1716a1.png)

to this, I can only have one bin (yielding Count == 1 values), when I specify manually a large bin size. But then there is the "Points outside range" popup:

![image](https://cloud.githubusercontent.com/assets/345210/21956949/fc86cd7e-da8b-11e6-8f41-c4db082ebe4d.png)

When I remove the Date Histogram Group-by the x-axis setting becomes weird.

**Comment by demonkoryu** [comment]
What about a graph that shows the last n datapoints? When using a CI system like Jenkins, your X axis denotes build number instead of time. It would be great to be able to graph the last n builds to show trends and not the builds in the last n days.

**Comment by torkelo** [comment]
@alexanderzobnin maybe time to revisit this so we can add table support. Would like it to work with as automatic as possible (which is tricky), some way we can detect table response.

**Comment by MarkRx** [comment]
+1 for graphs with non-time. With the introduction of mysql it is now possible to graph more complicated relationships along with relationships that are not centered around time. In my case I want to graph relationships of some metric against a discrete set of sequential values instead of time.

**Comment by MarkRx** [comment]
@torkelo I notice that there are lingering pieces of a "custom" xaxis mode where presumably one can define which field to use for it. What became of that? https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/axes_editor.ts#L46

**Comment by rtmaster** [comment]
+1

**Comment by tammytankian** [comment]
+1

**Comment by zeqk** [comment]
Any progress?
Thanks

**Comment by namirsab** [comment]
This would be nice to have

**Comment by hitwill** [comment]
+1

**Comment by jnicholls** [comment]
I'm all for this. I don't like that I have to return series data over-time just to be aggregated by Grafana when I want to show a bar chart of the data grouped by series...I can aggregate the data in the data source and be far more performant. But beyond that, I'm forced to return time points that fit within the range of the dashboard to avoid the "Data points out of time range" error :-/

**Comment by clugoimx** [comment]
Grafana seems to be obsessed with time. A good portion of my data is not time based. For now I am adding "Fake" time data in my sql queries to get it to graph. This is far from ideal. Especially since some graphs need to update only once per day while others can update along with the dashboard update interval. Now sure how I can upvote this request. Please consider this my vote. :)

**Comment by clugoimx** [comment]
I haven't seen anything that helps. SQL seems to be an afterthought in grafana.

**Comment by edum18** [comment]
> I haven't seen anything that helps. SQL seems to be an afterthought in grafana.

hey I found a workaround on a comment somewhere. basically just add the column "SYSDATETIME() as time" (on MSSQL) to your query select  and it will interpret as a time series, but the result is with no timeseries as we want.

**Comment by bassmaster187** [comment]
+1

**Comment by mscheer93** [comment]
I used following plugin as a workaround for non-time data:

https://grafana.com/plugins/natel-plotly-panel

**Comment by clugoimx** [comment]
Ooh, this looks interesting.

**Comment by georgewitteman** [comment]
This feature would be super awesome to use with Elasticsearch. Is anyone working on this?

**Comment by LuisGV212** [comment]
Is this still something in the works?

**Comment by f0ff886f** [comment]
Does anyone know of any alternatives? Realistically I have some monotonically increasing label (like a build number, for example, but perhaps it follows some versioning such as `year.buildnumber`) and would like to plot metrics for that.

Currently I'm leaning towards implementing something by hand, but it feels like in 2019 this kind of stuff has some prior art.

**Comment by jbg** [comment]
Select an additional column like `current_timestamp AS time` alongside your metric and value columns. Don't use that column for anything. Set the metric type to "time series", and then you will be able to choose Series for the x-axis mode and graph your data whichever way you want.

The fact that this has such a simple workaround makes it quite strange that it still can't just work the obvious way after all these years.

**Comment by MalteMagnussen** [comment]
Currently trying to Graph data with a column like this: 
![image](https://user-images.githubusercontent.com/11192971/94672767-0308bc00-0316-11eb-9090-ea7f2d40e2db.png)

How do I make a graph, that shows on the Y axis the number of rows up until that date? (Total number of users to that date)
And time on the X axis? (Based on this user_registered column)

For a tool to visualize data, it seems strange that this isn't an easy thing to do. Or maybe I'm just ignorant, and just need a guiding hand.

**Comment by MalteMagnussen** [comment]
> Select an additional column like `current_timestamp AS time` alongside your metric and value columns. Don't use that column for anything. Set the metric type to "time series", and then you will be able to choose Series for the x-axis mode and graph your data whichever way you want.
> 
> The fact that this has such a simple workaround makes it quite strange that it still can't just work the obvious way after all these years.

Could you make a guide for this? I can't quite seem to follow these instructions. Could you make an example with images and step by step? Or does it exist somewhere? Not sure what to google for.

**Comment by povi2946** [comment]
I need to plot FFT data.. X-Frequency, Y-Amplitude. We use grafana for many other sensor data. But it it's critical to have good frequency data shown too. It would be great if on X axis we could add annotations that are not based on timeseries.
So is this feature coming ? To plot a table.
